ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -336,6 +357,113 @@ function upgrade() {
         clear_upgrade
         check_upgrade_status "${cluster_src_ver}"
     fi
+    if check_ota_case_enabled "OCP-25473"; then
+        # As https://issues.redhat.com/browse/OTA-861 required, we can re-target to a z version when an upgrade is processing.
+        # For example, there is one upgrade is in progress A -> B (no matter it is a y stream or a z stream upgrade)
+        # Then:
+        # 1. we can retarget to a version which has same minor version with B
+        # 2. we can NOT retarget to a version which minor version is great than B
+        # *******************
+        # Nightly build is not applicable for z stream retarget tests
+        # *******************
+        # No matter y stream or z stream retarget, the TARGET always the final version we want to upgrade to
+        # To make OCP-25473 not block the whole pipeline, the upgrade orders are:
+        #   For z stream retarget upgrade: initial -> ${SHARED_DIR}/upgrade-edge -> TARGET, for example: 4.17/18.0 -> 4.18.1 -> 4.18.2 (this is the final target)
+        #   For y stream retarget upgrade: initial -> TARGET -> ${SHARED_DIR}/upgrade-edge, for example: 4.17/18.0 -> 4.18.1 (this is the final target) -> 4.19.0
```

Comment:
this is a bit of a tricky part. it means that whenever we want to test y-stream retarget, we'll need to wait for the next version to have at least ec.0
question is, what we prefer doing when we don't have this version yet? do we just skip the test? or make the ci to delay running this profile until we have such version? in any case i guess we definitely don't want it to fail just because no such version yet.

Reply:
i think this is the most important one

Reply:
I"ve seen you just done it manually in a way, by only creating 16 to 17 job for now.. so the plan is to add 18 once 19ec0 is available? and each new version manually "delay" the addition of it to ci?

Reply:
says right now we are on 4.18, then we only add retarget y stream tests on 4.16. then when we have 4.19, we add  retarget y stream tests for 4.17.
It seems there is no way to check current version for y stream retarget.

Reply:
btw plz note that even though I was testing the fix on 4.17, the merge target for the PR is 4.18. it is unclear to me if there will be a backport to 4.17, which I doubt since the ticket is of a story type rather than a bug

Reply:
> says right now we are on 4.18, then we only add retarget y stream tests on 4.16. then when we have 4.19, we add retarget y stream tests for 4.17. It seems there is no way to check current version for y stream retarget.

yep. basically your approach is the obvious one. the only caveat is we'll have to remember adding it just on time. the time window we have from the moment 4.19ec0 is available, to the moment 4.18 is released is not very long, and we should remember to update this job to new version as close to the time next ec0 is available as possible, to avoid releasing without a proper y retarget test. ideally, we should have some sort of automatic skip condition whenever there's no release available yet, so we're not dependent on manual work each release just to keep this test working

Reply:
I think that's not a big deal because we can copy a new ota config file and I think it is the time to copy it as well.
cc @jiajliu can we copy a new ota config file? openshift-verification-tests-master__ota-multi-stable-4.18-upgrade-from-stable-4.17.yaml

Reply:
we can test 4.17 -> 4.17 -> 4.18 to test y stream retarget.

Reply:
we can test 4.17 -> 4.17 -> 4.18 to test y stream retarget.

Reply:
thing is, each release we will be limiting ourselves to copying the ota file "just in the right time" which is "after next ec.0" and "before ge" at the same time. 
luckily, I guess we don't have to rely on `ystreamretarget` to be selected by prow. since we know ec.0 is always the first version to arrive, we can just check for that version in the code. if its not available yet, skip the test.

Reply:
> we can test 4.17 -> 4.17 -> 4.18 to test y stream retarget.

well not really. while you definitely can test that, this test will verify 4.17 but not 4.18

Reply:
the code being "tested" is always in the "middle" version

Reply:
I think we actually just need to copy a new file in advance.  
It seems we are missed a 4.18 file for current version :)

Reply:
> well not really. while you definitely can test that, this test will verify 4.17 but not 4.18
Ok, so we actually can never test latest version :(

Reply:
> I think we actually just need to copy a new file in advance.

yep. but we will need a skip condition in the code, so the test this new file adds will not fail the first time when the next ec0 is not ready yet.

>  Ok, so we actually can never test latest version :(

i think we actually can, we just need to wait for the next ec.0 it should be available very soon i think, before 4.18 is GA

Reply:
timing of available versions is like that:
1) 4.18 nightly
2) 4.18 EC0
3) 4.17 GA
4) 4.19 nightly
5) 4.19 EC0
6) 4.18 GA
...

with that, the time we have to test z retarget is anywhere between 2 to 6, a lot of time. the time we have for Y is between 5 to 6, a short time (was around 20 days for 4.17).. we're now in 4..

Reply:
> but we will need a skip condition in the code, so the test this new file adds will not fail the first time when the next ec0 is not ready yet.
If the stable is not ready and we add it config file will cause below error:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58136/rehearse-58136-periodic-ci-openshift-verification-tests-master-ota-multi-stable-4.18-upgrade-from-stable-4.17-baremetal-ipi-ovn-ipv4-arm-f28/1861629692935671808